ndcube/utils/wcs.py

```diff
@@ -285,42 +285,82 @@ def reindex_wcs(wcs, inds):
     return outwcs
 
 
-def get_dependent_axes(wcs_object, axis, missing_axis):
-    # Given an axis number in numpy ordering, returns the axes whose
-    # WCS translations are dependent, including itself.  Again,
-    # returned axes are in numpy ordering convention.
-    # Copied from WCSCoordinates class in glue-viz/glue github repo.
+def get_dependent_data_axes(wcs_object, data_axis, missing_axis):
+    """
+    Given a data axis index, return indices of dependent data axes.
 
-    # TODO: we should cache this
+    Both input and output axis indices are in the numpy ordering convention
+    (reverse of WCS ordering convention). The returned axis indices include the input axis.
+    Returned axis indices do NOT include any WCS axes that do not have a
+    corresponding data axis, i.e. "missing" axes.
 
-    # if distorted, all bets are off
-    try:
-        if any([wcs_object.sip, wcs_object.det2im1, wcs_object.det2im2]):
-            return tuple(range(wcs_object.naxis))
-    except AttributeError:
-        pass
+    Parameters
+    ----------
+    wcs_object: `astropy.wcs.WCS` or `ndcube.utils.wcs.WCS`
+        The WCS object describing the axes.
+
+    data_axis: `int`
+        Index of axis (in numpy ordering convention) for which dependent axes are desired.
+
+    missing_axis: iterable of `bool`
+        Indicates which axes of the WCS are "missing", i.e. do not correspond to a data axis.
+
+    Returns
+    -------
+    dependent_data_axes: `tuple` of `int`
+        Sorted indices of axes dependent on input data_axis in numpy ordering convention.
+
+    """
+    # In order to correctly account for "missing" axes in this process,
+    # we must determine what axes are dependent based on WCS axis indices.
+    # Convert input data axis index to WCS axis index.
+    wcs_axis = utils_cube.data_axis_to_wcs_axis(data_axis, missing_axis)
+    # Determine dependent axes, including "missing" axes, using WCS ordering.
+    wcs_dependent_axes = np.asarray(get_dependent_wcs_axes(wcs_object, wcs_axis, missing_axis))
+    # Remove "missing" axes from output.
+    non_missing_wcs_dependent_axes = wcs_dependent_axes[
+        np.invert(missing_axis)[wcs_dependent_axes]]
+    # Convert dependent axes back to numpy/data ordering.
+    dependent_data_axes = tuple([utils_cube.wcs_axis_to_data_axis(i, missing_axis)
+                                 for i in non_missing_wcs_dependent_axes])
+    return dependent_data_axes
+
+
+def get_dependent_wcs_axes(wcs_object, wcs_axis, missing_axis):
```

Comment:
is `missing_axis` actually being used in this function?

Reply:
No.  That's for the spot.

Just working on tests now.